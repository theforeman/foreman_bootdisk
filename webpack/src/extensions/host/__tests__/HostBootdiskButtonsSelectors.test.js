import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';

import {
  selectHostId,
  selectIsLoading,
  selectBootdiskOptions,
} from '../HostBootdiskButtonsSelectors';

import {
  HOST_DETAILS_REQUEST_KEY,
  HOST_BOOTDISK_BUTTONS_REQUEST_KEY,
} from '../../constants';

const stateFactory = (bootdiskState, hostState) => ({
  API: {
    [HOST_DETAILS_REQUEST_KEY]: {
      response: {
        ...hostState,
      },
    },
    [HOST_BOOTDISK_BUTTONS_REQUEST_KEY]: {
      response: {
        ...bootdiskState,
      },
    },
  },
});

const state = stateFactory(
  {
    bootdiskOptions: {
      bootdiskDownloadable: true,
      architectureName: 'x86_64',
      actions: [
        {
          title: "Host 'sppup' image",
          link: '/bootdisk/disks/hosts/8',
          disabled: false,
          description: null,
        },
        {
          title: "Full host 'sppup' image",
          link: '/bootdisk/disks/full_hosts/8',
          disabled: true,
          description: 'Host is not in build mode',
        },
        {
          title: 'Boot disk help',
          link: '/bootdisk/disks/help',
        },
      ],
    },
  },
  {
    id: 8,
  }
);

const fixtures = {
  'should return options': () => selectBootdiskOptions(state),
  'should return isLoading': () => selectIsLoading(state),
  'should return hostId': () => selectHostId(state),
};

describe('HostBootdiskButtons selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
