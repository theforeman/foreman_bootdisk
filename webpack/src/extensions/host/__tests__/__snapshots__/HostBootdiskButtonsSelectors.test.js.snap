// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostBootdiskButtons selectors should return hostId 1`] = `8`;

exports[`HostBootdiskButtons selectors should return isLoading 1`] = `true`;

exports[`HostBootdiskButtons selectors should return options 1`] = `
Object {
  "actions": Array [
    Object {
      "description": null,
      "disabled": false,
      "link": "/bootdisk/disks/hosts/8",
      "title": "Host 'sppup' image",
    },
    Object {
      "description": "Host is not in build mode",
      "disabled": true,
      "link": "/bootdisk/disks/full_hosts/8",
      "title": "Full host 'sppup' image",
    },
    Object {
      "link": "/bootdisk/disks/help",
      "title": "Boot disk help",
    },
  ],
  "architectureName": "x86_64",
  "bootdiskDownloadable": true,
}
`;
